cmd/sonobuoy/app/images.go

```diff
@@ -0,0 +1,293 @@
+}
+
+func ImagesFlagSet(cfg *imagesFlags) *pflag.FlagSet {
+	flagset := pflag.NewFlagSet("images", pflag.ExitOnError)
+	AddPluginFlag(&cfg.plugin, flagset)
+	AddE2ERegistryConfigFlag(&cfg.e2eRegistryConfig, flagset)
+	AddRegistryUsernameFlag(&cfg.username, flagset)
+	AddKubeconfigFlag(&cfg.kubeconfig, flagset)
+	cfg.imagesflags = flagset
+	return flagset
+}
+
+func NewCmdImages() *cobra.Command {
+	// Main command
+	cmd := &cobra.Command{
+		Use:   "images",
+		Short: "Manage images used in a plugin. Supported plugins are: 'e2e'",
+		Run:   getImages,
+		Args:  cobra.ExactArgs(0),
+	}
+
+	cmd.Flags().AddFlagSet(ImagesFlagSet(&imagesflags))
+	cmd.MarkFlagRequired(pluginFlag)
+
+	// Sub command
+	subCmd := &cobra.Command{
+		Use:   "download",
+		Short: "Download images to local Docker client for a specific plugin",
+		Run:   downloadImages,
+		Args:  cobra.ExactArgs(0),
+	}
+	subCmd.Flags().AddFlagSet(ImagesFlagSet(&imagesflags))
+	subCmd.MarkFlagRequired(pluginFlag)
+	subCmd.MarkFlagRequired(e2eRegistryConfigFlag)
+
+	cmd.AddCommand(subCmd)
+
+	return cmd
+}
+
+func getImages(cmd *cobra.Command, args []string) {
+
+	if len(imagesflags.e2eRegistryConfig) > 0 {
+		// Check if the e2e file exists
+		if _, err := os.Stat(imagesflags.e2eRegistryConfig); os.IsNotExist(err) {
+			errlog.LogError(errors.Errorf("file does not exist or cannot be opened: %v", imagesflags.e2eRegistryConfig))
+			os.Exit(1)
+		}
+	}
+
+	switch imagesflags.plugin {
+	case "e2e":
+
+		cfg, err := imagesflags.kubeconfig.Get()
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't get REST client"))
+			os.Exit(1)
+		}
+
+		sbc, err := getSonobuoyClient(cfg)
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "could not create sonobuoy client"))
+			os.Exit(1)
+		}
+
+		version, err := sbc.Version()
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't get Sonobuoy client"))
+			os.Exit(1)
+		}
+
+		// Get list of images that match the version
+		registry, err := image.NewRegistryList(imagesflags.e2eRegistryConfig, version)
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't init Registry List"))
+			os.Exit(1)
+		}
+
+		images, err := registry.GetImageConfigs()
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't get images for version"))
+			os.Exit(1)
+		}
+
+		for _, v := range images {
+			fmt.Println(v.GetE2EImage())
+		}
+	default:
+		errlog.LogError(errors.Errorf("Unsupported plugin: %v", imagesflags.plugin))
+		os.Exit(1)
+	}
+}
+
+func downloadImages(cmd *cobra.Command, args []string) {
+
+	// Check if the e2e file exists
+	if _, err := os.Stat(imagesflags.e2eRegistryConfig); os.IsNotExist(err) {
+		errlog.LogError(errors.Errorf("file does not exist or cannot be opened: %v", imagesflags.e2eRegistryConfig))
+		os.Exit(1)
+	}
+
+	// Check if username if specified for Registry Auth
+	if len(imagesflags.username) > 0 {
+		fmt.Print("Registry password: ")
+		bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
```

Comment:
ENV isn't really a secure place to put stuff which is why I had it prompt, it's never written anywhere. Could add though if you feel strongly we need it as ENV.

Reply:
I was under the impression (as a non-security expert) that the env was a _good_ place to put creds. Not that "other people doing it" makes it right, but:

 - [AWS](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) does it for their cli and other tools. They do cite that it is particularly useful for scripting and other use cases though.
 - [SauceLabs](https://wiki.saucelabs.com/display/DOCS/Best+Practice%3A+Use+Environment+Variables+for+Authentication+Credentials) lists is as a best practice

ENV isn't providing security per-se since it is unencrypted but the alternative for a lot of workflows (e.g. CI or Docker) would be setting the values as env values and then (if you require them on stdin) they'd have to just echo them through a pipe and hope that is handled correctly (I've seen password libraries including the go stdlib have problems on different platforms w.r.t. pipes getting broken or not being handled at all or causing other issues).

I guess in some situations you're right though: if I'm a user standing in front of the keyboard, _having_ to write the password to the env would be less secure since it would be visible. They _could_ write it to a file and source it or something like that, but its non-ideal for any user.

Thoughts on the dual support or do you think that is still actively a bad idea to support at all?